web_src/js/utils/dom.js

```diff
@@ -227,3 +227,20 @@ export function initSubmitEventPolyfill() {
   document.body.addEventListener('click', submitEventPolyfillListener);
   document.body.addEventListener('focus', submitEventPolyfillListener);
 }
+
+/**
+ * Check if an element is visible.
+ * Note: This function doesn't account for all possible visibility scenarios.
+ * @param {HTMLElement} element The element to check.
+ * @returns {boolean} True if the element is visible.
+ */
+export function isVisible(element) {
+  if (!element) return false;
+
+  const style = window.getComputedStyle(element);
+  return style.width !== '0' &&
+           style.height !== '0' &&
+           style.opacity !== '0' &&
+           style.display !== 'none' &&
+           style.visibility !== 'hidden';
+}
```

Comment:
Maybe just make it match jQuery instead?

https://github.com/jquery/jquery/blob/bf11739f6c6926bc9bc1b5a1460505d3b7ef8b01/src/css/hiddenVisibleSelectors.js#L8-L10

Reply:
Done https://github.com/go-gitea/gitea/pull/29271/commits/f695a012f2169b7fd2843894e19a35592eea913c